ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyOptimizer.java

```diff
@@ -435,15 +439,39 @@ private void transitionLayoutOnlyViewToNativeView(
     mTagsWithLayoutVisited.clear();
   }
 
-  private static boolean isLayoutOnlyAndCollapsable(@Nullable ReactStylesDiffMap props) {
+  private static boolean isCollapsableChildren(@Nullable ReactStylesDiffMap props) {
     if (props == null) {
       return true;
     }
 
+    if (props.hasKey(ViewProps.COLLAPSABLE_CHILDREN) && !props.getBoolean(ViewProps.COLLAPSABLE_CHILDREN, true)) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private static boolean isCollapsable(@Nullable ReactStylesDiffMap props) {
+    if (props == null) {
+      return false;
```

Comment:
This should return true for consistency with collapsableChildren (and to maintain current behavior)